org.adempiere.base/src/org/compiere/model/GridTable.java

```diff
@@ -645,7 +647,8 @@ public boolean open (int maxRows)
 			m_buffer = new ArrayList<Object[]>(m_rowCount+10);
 		}
 		m_sort = new ArrayList<MSort>(m_rowCount+10);
-		if (m_rowCount > 0)
+		//actual row count or -1 for timeout

```

Comment:
I leave a wrong comments here, should be just "//actual row count or timeout"

Reply:
fixed